src/validator.cpp

```diff
@@ -679,10 +679,10 @@ void Validator::ValidatorImpl::validateVariable(const VariablePtr &variable, con
         mValidator->addError(err);
     } else if (!isStandardUnitName(variable->units())) {
         ComponentPtr component = std::dynamic_pointer_cast<Component>(variable->parent());
-        ModelPtr model = std::dynamic_pointer_cast<Model>(component->parent());
+        ModelPtr model = owningModel(component);
```

Comment:
What's the rationale for having a separate function `owningModel` outside the Entity class? It's not really consistent with other behaviour?

Reply:
The `owningModel` function has nothing to do with the `Entity` class.  It is merely a utility function that saves a little bit of work.

Reply:
... and it shouldn't it be included in the Entity class because ...?

Reply:
The `Entity` class should know nothing of  the `Model` class.